ui/public/checkBrowser.js

```diff
@@ -0,0 +1,45 @@
+// Consider this js file must run normally in the old browsers likes IE 6,
+// so we can't use the new grammars and APIs (likes let/const, string interpolator, querySelector, etc) in this file.
+// We need to handle the compatibility carefully.
+
+function browserLang() {
+  // https://zzz.buzz/2016/01/13/detect-browser-language-in-javascript/
+  // https://social.msdn.microsoft.com/Forums/sqlserver/en-US/a3d78aaf-9f70-4826-954d-19183173c1c3/how-to-change-navigatoruserlanguage-in-ie11
+  return (
+    (navigator.languages && navigator.languages[0]) ||
+    navigator.language ||
+    navigator.browserLanguage ||
+    navigator.userLanguage ||
+    'en'
+  )
+}
+
+function checkBrowser(supportedBrowsers) {
+  if (supportedBrowsers.test(navigator.userAgent)) {
+    var text
+    if (browserLang().indexOf('zh') === 0) {
+      text =
+        '一些功能在此浏览器上可能无法工作，请使用最新版本的 Chrome/Edge/Firefox/Safari 浏览器。'
+    } else {
+      text =
+        'Some features may not work in your browser. Please use latest Chrome/Edge/Firefox/Safari browsers.'
```

Comment:
I think it is enough to use `latest` to represent the version instead of a specific version value. Because when people want to upgrade or download an app, they just care about the newest version. How do you think about it? @breeswish

Reply:
Yes, I think so!